### PR TITLE
Copilot Chat: Update attributes on CopilotChat skills

### PR DESCRIPTION
**Pull request by gitri-ms**
### Motivation and Context

The SK attribute format we were using on the Copilot Chat skills is being deprecated. 

### Description

This change updates the attributes to the new format and make the warnings go away. 

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#dev-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/SemanticChatMemorySkill.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public class SemanticChatMemorySkill
     /// <param name="context">The SKContext</param>
     /// <param name="query">Query to match.</param>
     /// <returns>A string containing the relevant memories.</returns>
-    [SKFunction("Query chat memories")]
+    [SKFunction, Description("Query chat memories")]
     public async Task<string> QueryMemoriesAsync(
         SKContext context,
```
Comment (shawncal) [inline samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/SemanticChatMemorySkill.cs:42]: Remove context (and /// <param>) -- unused

File: samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/DocumentMemorySkill.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ public class DocumentMemorySkill
     /// </summary>
     /// <param name="query">Query to match.</param>
     /// <param name="context">The SkContext.</param>
```
Comment (shawncal) [inline samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/DocumentMemorySkill.cs:49]: Update /// <params> to match args please :)

Comment (shawncal) [reply samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/DocumentMemorySkill.cs:49]: (I'm also fine just deleting args and keeping only the <summary> for the non-SDK code, as long that doesn't throw warnings)

File: samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/DocumentMemorySkill.cs [hunk 3]
```diff
@@ -41,13 +41,16 @@ public class DocumentMemorySkill
     /// Query the document memory collection for documents that match the query.
     /// </summary>
     /// <param name="query">Query to match.</param>
-    /// <param name="context">The SkContext.</param>
+    /// <param name="chatId">ID of the chat that owns the documents.</param>
+    /// <param name="tokenLimit">Maximum number of tokens.</param>
+    /// <param name="textMemory">Semantic text memory.</param>
+    /// <returns>A string containing snippets from the relevant documents.</returns>
     [SKFunction, Description("Query documents in the memory given a user message")]
     public async Task<string> QueryDocumentsAsync(
```
Comment (gitri-ms) [inline samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/DocumentMemorySkill.cs:49]: Need to update error handling when this is called. Currently checking context.ErrorOccurred, which is not populated when called directly as a normal C# function (as opposed to being called by the kernel)

File: samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/SemanticChatMemorySkill.cs [hunk 4]
```diff
@@ -34,14 +34,16 @@ public class SemanticChatMemorySkill
...
     /// </summary>
     /// <param name="query">Query to match.</param>
-    /// <param name="context">The SKContext</param>
+    /// <param name="chatId">Chat ID to query history from.</param>
+    /// <param name="tokenLimit">Maximum number of tokens.</param>
+    /// <param name="textMemory">Semantic text memory.</param>
     /// <returns>A string containing the relevant memories.</returns>
     [SKFunction, Description("Query chat memories")]
     public async Task<string> QueryMemoriesAsync(
         [Description("Query to match.")] string query,
```
Comment (glahaye) [inline samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/SemanticChatMemorySkill.cs:43]: I don't believe you need the attributes on each parameter as the descriptions are already extracted from the triple-slash comments

File: samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/ChatSkill.cs [hunk 5]
```diff
@@ -182,7 +182,6 @@ public async Task<string> ExtractAudienceAsync(SKContext context)
     /// <summary>
```
Comment (glahaye) [inline samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/ChatSkill.cs:182]: Will this, or has this been sent to the new CC repo?

File: samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/SemanticChatMemorySkill.cs (hunk 4 above)
Comment (shawncal) [reply samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/SemanticChatMemorySkill.cs:43]: Descriptions are not extracted from comments. But you only need the [Description] on params that the AI could set (via ContextVariables). If they're being set deterministically by the Kernel (i.e. CancellationToken), then Description isn't used.

Note: ISemanticTextMemory has been removed from the SKContext and will no longer be auto-mapped to fill this arg as it once was. I recommend you pass in an ISemanticTextMemory instance in to the skill constructor instead.

**Comment by shawncal** [comment]
@gitri-ms Closing as this will need to be ported to the new repo.

**Pull request closed without merging**